### PR TITLE
added RO types to replace any types, add try catch and toast in service

### PR DESCRIPTION
**Pull request by coryatwater-FW**

### CODE REVIEW

File: apps/web/src/components/milestone-logs/recruitment/Record.tsx [hunk 1]
```diff
@@ -90,7 +81,7 @@ export const Record: React.FC<RecordProps> = ({ job }) => {
               <DetailsItem title='Recruiter Name' text={recruiter_name} />
               <DetailsItem
                 title='Date Job Was First Posted'
-                text={formatDate(job_post_date.toString())}
+                text={job_post_date && formatDate(job_post_date)}
```
Comment (sunghwan-park-fw) [inline apps/web/src/components/milestone-logs/recruitment/Record.tsx:84]: formatDate might handle null and undefined.

Comment (coryatwater-FW) [reply apps/web/src/components/milestone-logs/recruitment/Record.tsx:84]: you are correct, I will remove those checks

File: apps/web/src/components/milestone-logs/recruitment/Milestone.tsx [hunk 2]
```diff
@@ -46,14 +45,10 @@ export const AddMilestones: React.FC<AddMilestoneProps> = ({
     values.job_id = jobId.toString();
     values.added_by = '1';
 
-    try {
-      const {
-        data: { data },
-      } = await addMilestone(applicantId as string, values);
+    const data = await addMilestone(applicantId as string, values);
```
Comment (sunghwan-park-fw) [inline apps/web/src/components/milestone-logs/recruitment/Milestone.tsx:48]: applicantId's type is `string | string[] | undefined`. should we change it to `string` or check if it is string here?

Comment (coryatwater-FW) [reply apps/web/src/components/milestone-logs/recruitment/Milestone.tsx:48]: changed it to string in parent comp so we can change type to just string

File: packages/common/src/helper/jsonToArray.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// used for converting json values to an array to allow map fn to work
+export const jsonToArray = (json?: any) => {
```
Comment (sunghwan-park-fw) [inline packages/common/src/helper/jsonToArray.ts:2]: isn't it same with Object.values(json)?

**Review by sunghwan-park-fw — APPROVED** [review]
I left some comments.

File: packages/common/src/helper/jsonToArray.ts (hunk 3 above)
Comment (coryatwater-FW) [reply packages/common/src/helper/jsonToArray.ts:2]: yes, I changed it to Object.values since I have no need for the keys, good catch!

**Pull request merged**